Updating the production node install Cardona requirements (#475)
0xPolygon/polygon-docs: docs/zkEVM/get-started/setup-nodes/production-node.md

comments: true
---

The Polygon zkEVM beta mainnet is available for developers to launch smart contracts, execute transactions, and experiment. This document shows you how to launch your own production zkNode.

Developers can setup a production node with either the Polygon zkEVM mainnet or the Cardona testnet.

After spinning up an instance of the production node, you can run the synchronizer and utilize the JSON-RPC interface.

!!!info
    - Sequencer and prover functionalities are not covered in this document as they are still undergoing development and rigorous testing.
    - Syncing the zkNode currently takes anywhere between 1-2 days depending on various factors. The team is working on snapshots to improve the syncing time.

## Prerequisites

This tutorial requires a [`docker-compose`](https://docs.docker.com/compose/install/) installation.

> Run the following to create directory:

Author: ```suggestion
Run the following to create a directory:
```

```sh
mkdir -p ~/zkevm-node
```

### Minimum hardware requirements

!!!caution
    - The zkProver does not work on ARM-based Macs yet. 
    - For Windows users, the use of WSL/WSL2 is not recommended.
    - Currently, zkProver optimizations require CPUs that support the AVX2 instruction, which means some non-M1 computers, such as AMD, won't work with the software regardless of the OS.

- 16GB RAM
- 4-core CPU
- ~250/350GB storage

### Software requirements

- An Ethereum node; Geth or any service providing a JSON RPC interface for accessing the L1 network.
- zkEVM node (or zkNode) for the L2 network.
- Synchronizer which is responsible for synchronizing data between L1 and L2.
- A JSON RPC server which acts as an interface to the L2 network.

## Ethereum node setup

We set up the Ethereum node first as it takes a long time to synchronize.

> There are numerous ways to set up an Ethereum L1 environment; we will use Geth for this. We recommend Geth, but a Sepolia node is OK too.

Author: ```suggestion
We recommend using Geth but a Sepolia node is OK too.
```

4. The `example.env` file must be modified according to your configurations.

>         Edit the .env file with your favourite editor (we use nano): 

Author: ```suggestion
        Edit the .env file with your favorite editor (we use nano): 
```

    ```bash
    docker compose --env-file $ZKEVM_CONFIG_DIR/.env -f $ZKEVM_DIR/$ZKEVM_NET/docker-compose.yml logs <cointainer_name>
    ```

!!!info "Batch rate"
    - Batches are closed every 10s, or whenever they are full.

>     - Also, the frequency of closing batches is subject to change as it depends on the prevailing configurations.

Author: ```suggestion
    - The frequency of closing batches is subject to change as it depends on the prevailing configurations.
```
